browser/widevine/widevine_utils.cc

```diff
@@ -130,10 +135,38 @@ void RegisterWidevineProfilePrefsForMigration(
 }
 
 void RegisterWidevineLocalstatePrefs(PrefRegistrySimple* registry) {
+  // On Arm64 Linux, we would actually like the default value to be
+  // HasBundledWidevine(). But it's not possible to call this function here.
+  // So we set the default value to false here and call HasBundledWidevine()
+  // further below.
   registry->RegisterBooleanPref(kWidevineEnabled, false);
 }
 
+// On Arm64 Linux (and other non-x64 architectures), Widevine is not publicly
+// available. Users may obtain it themselves and place it into Brave's
+// installation directory. HasBundledWidevine() checks if this has happened:
+#if BUILDFLAG(IS_LINUX) && !defined(ARCH_CPU_X86_64)
+bool HasBundledWidevine() {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  content::CdmRegistryImpl* cdm_registry =
+      content::CdmRegistryImpl::GetInstance();
+  CHECK(cdm_registry);
+  const std::vector<content::CdmInfo>& cdms = cdm_registry->GetRegisteredCdms();
+  base::FilePath bundled_dir =
+      base::PathService::CheckedGet(chrome::DIR_BUNDLED_WIDEVINE_CDM);
+  for (auto it : cdms) {
+    if (it.key_system == kWidevineKeySystem && bundled_dir.IsParent(it.path)) {
```

Comment:
Just curious `bundled_dir.IsParent(it.path)` is necessary here. checking `it.key_system` isn't sufficient?

Reply:
In general, `cdm_registry->GetRegisteredCdms()` also contains component-updated Widevine. The `IsParent` check is needed to distinguish a component-updated Widevine from a bundled one.

On the other hand, you are right that at the moment, on non-x64 Linux where there is currently no component-updated Widevine, the `IsParent` is not necessary. But maybe in the future there will be component-updated Widevine on Arm64 Linux. Then, the `IsParent` check will matter.